src/main/scala/edu/berkeley/ce/sparkrocks/SeedJointSelector.scala

```diff
@@ -3,242 +3,221 @@ package edu.berkeley.ce.sparkrocks
 import scala.annotation.tailrec
 
 /**
-  * Selects joints from the input joint set that will be used to seed the RDD in order to maintain
-  * approximately balanced loads among processors.
+  * Selects joints from the input joint set that will be used to seed the RDD in order to
+  * maintain approximately balanced loads among processors.
   */
 object SeedJointSelector {
-  val THRESHOLD = 0.50
+  // Threshold for joint acceptance
+  val THRESHOLD = 0.1
 
   /**
-    * Searches though sequence of joint sets to find best seed joints
+    * Generate specified number of partitions using input joint sets
     *
     * @param jointSets Seq of Seq's of Joints representing input joint sets
     * @param inputVolume Block representing input rock volume
-    * @param numProcessors Number of processors to be used in the analysis
-    * @param remainingJoints Seq of Seq of joints that were not selected as seed joints
-    * @return Returns a tuple containing two Seq's of Joints. The first Seq contains the seed joints that
-    *         best divide the input volume. The second Seq contains the joints that remain from the input
-    *         joint set.
+    * @param numPartitions Number of partitions desired
+    * @param jointSetSpan Number of joint sets to search at a time for partition generation
+    * @param seedBlocks Blocks that will serve as partitions
+    * @param remainingJoints Seq of joints that were not used in seed block generation
+    * @return Returns a tuple containing a Seq of blocks and a Seq of joints. The Seq of blocks
+    *         are the the desired partitions. The Seq of joints are the remaining joints that were
+    *         not used to generate the partitions.
     */
   @tailrec
-  def searchJointSets(jointSets: Seq[Seq[Joint]], inputVolume: Block, numProcessors: Int,
-                      remainingJoints: Seq[Joint] = Seq.empty[Joint]): (Seq[Joint], Seq[Joint]) = {
-    if (jointSets.isEmpty) {
-      throw new IllegalStateException("ERROR: Unable to find satisfactory seed joints")
-    }
-    // Check that joint set is persistent
-    if (jointSets.head(0).shape.isEmpty) {
-      val seedJointOption = findSeedJoints(jointSets.head, inputVolume, numProcessors, inputVolume.volume)
-
-      // Check whether joint set yields satisfactory seed joints. If it does not, check next joint set.
-      if (seedJointOption.isEmpty) {
-          searchJointSets(jointSets.tail, inputVolume, numProcessors, remainingJoints ++ jointSets.head)
+  def generateSeedBlocks(jointSets: Seq[Seq[Joint]], inputVolume: Block, numPartitions: Int,
+                         jointSetSpan: Int = 1, seedBlocks: Seq[Block] = Seq.empty[Block],
+                         remainingJoints: Seq[Joint] = Seq.empty[Joint]): (Seq[Block], Seq[Joint]) = {
```

Comment:
I understand why `findSeedBlocks` has a `remainingJoints` parameter, but it isn't clear to me what that parameter is really doing in this function.

It seems to me that if you fail to find the specified number of partitions when `jointSetSpan = n`, you don't really care about the joints that were left out from this attempt when moving on to try `jointSetSpan = n + 1`.